### ISSUE TITLE
Comments in json target

### ISSUE THREAD

**Issue by adematte** [dcodeIO/protobuf.js#1145]
protobuf.js version: 6.8.8

Currently when using pbjs with the json target flag `-t json`, all comments are stripped from the output file and there is no option to preserve comments. Preserving comments however would be a great way to generate documentation from the json file.

I propose adding two new flags to pbjs 
- keep-comments (defaults to false)
- alternate-comment-mode (defaults to false)

These parameters are the same as those used throughout the code and would affect a few lines of code in protobufjs/cli/pbjs.js and most notably:
```js
    var parseOptions = {
        "keepCase": argv["keep-case"] || false,
        "keepComments": argv["keep-comments"] || false, // new to preserve comments
        "alternateCommentMode": argv["alternate-comment-mode"] || false, // new option to parse alternate comments
    };
```
Would this make sense? I can submit a PR quickly upon proposal approval.

**Comment by WanderWang** [comment]
I had the same problem . I modify  ```protobufjs/cli/target/json.js```  in my node_modules temporary


```
"use strict";
module.exports = json_target;

json_target.description = "JSON representation";

function json_target(root, options, callback) {
    console.log(options)
    callback(null, JSON.stringify(root.toJSON({ keepComments: true }), null, 2)); //hard code keepComments
}
```

**Comment by adematte** [comment]
@WanderWang yes I actually used to do the same thing except that I did not modify protobufjs/cli/target/json.js but instead created a new target: did you know you can pass a custom target like this `-t my-json-target.js`? It's better than hacking your node_modules ;-)

However just having a custom target doesn't seem to work anymore since >6.8.6 if I remember correctly. The comments are no longer parsed before reaching the target.

**Comment by phantomk** [comment]
I had the same problem. has better solution?

**Comment by mathewbyrne** [comment]
Yep, I've had to manage a custom CLI script for generating JSON outputs to keep comments that use the alternative (and IMO more intuitive) comments mode.  It would be really neat if this could be included in the CLI package by default.

Since there's been no movement on this for a while I will look at submitting a PR with this change.

**Comment by wenerme** [comment]
@mathewbyrne do you opensourced your custom cli ? want to get the comments too.

**Comment by dcodeIO** [comment]
Closing this issue because it has been inactive for more than two years and the project has changed substantially.

If this is still relevant, please open a fresh issue with current reproduction steps, version details, and any updated context.

Please also consider that maintaining protobuf.js takes time and effort. If the project is valuable to you or your organization, sponsorship is appreciated.

Thanks!
